File: examples/src/main/java/aug/script/examples/java/AbstractClient.java
package aug.script.examples.java;

import aug.script.framework.ClientInterface;
import aug.script.framework.LineEvent;
import aug.script.framework.ProfileInterface;
import aug.script.framework.ReloadData;
import org.mongodb.scala.MongoClient;
import org.mongodb.scala.MongoDatabase;

public class AbstractClient implements ClientInterface {

    protected static ProfileInterface profile;

    @Override
    public void init(ProfileInterface profileInterface, ReloadData reloadData) {
        profile = profileInterface;
    }

    @Override
    public void initDB(MongoClient mongoClient, MongoDatabase mongoDatabase) {
    }

    @Override
    public ReloadData shutdown() {
        return null;
    }

    @Override
    public boolean handleLine(LineEvent lineEvent) {
        return false;
    }

    @Override
    public void handleFragment(LineEvent lineEvent) {

    }

    @Override
    public void handleGmcp(String gmcp) {

    }

    @Override
    public boolean handleCommand(String cmd) {
        return false;
    }

    @Override
    public void onConnect(long id, String url, int port) {

    }

    @Override
    public void onDisconnect(long id) {

    }
}

File: examples/src/main/scala/aug/script/examples/scala/AbstractClient.scala
package aug.script.examples.scala

import aug.script.framework.{ClientInterface, LineEvent, ProfileInterface, ReloadData}
import org.mongodb.scala.{MongoClient, MongoDatabase}

abstract class AbstractClient extends ClientInterface {
  protected var profile: ProfileInterface

  override def init(profileInterface: ProfileInterface, reloadData: ReloadData): Unit = {
    this.profile = profileInterface
  }

  override def shutdown(): ReloadData = { new ReloadData }

  override def handleLine(lineEvent: LineEvent): Boolean = { false }

  override def handleFragment(lineEvent: LineEvent): Unit = {}

  override def handleGmcp(gmcp: String): Unit = {}

  override def handleCommand(cmd: String): Boolean = { false }

  override def onConnect(id: Long, url: String, port: Int): Unit = {}

  override def onDisconnect(id: Long): Unit = {}

  override def initDB(mongoClient: MongoClient, mongoDatabase: MongoDatabase): Unit = {}
}
